File: ruby/hackerrank/utopian-tree.rb
#!/bin/ruby

require 'json'
require 'stringio'

# Complete the utopianTree function below.
def utopianTree(n)
    i = 1
    r = 1
    while i <= n
        if i % 2 == 1
            r *= 2
        else
            r += 1
        end
        i += 1
    end
    return r
end

fptr = File.open(ENV['OUTPUT_PATH'], 'w')

t = gets.to_i

t.times do |t_itr|
    n = gets.to_i

    result = utopianTree n

    fptr.write result
    fptr.write "\n"
end

fptr.close()

File: python/hackerrank/utopian-tree.py
#!/bin/python3

import math
import os
import random
import re
import sys

# Complete the utopianTree function below.
def utopianTree(n):
    i = 1
    r = 1
    while i <= n:
        if i % 2 == 1:
            r *= 2
        else:
            r += 1
        i += 1
    return r

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    t = int(input())

    for t_itr in range(t):
        n = int(input())

        result = utopianTree(n)

        fptr.write(str(result) + '\n')

    fptr.close()
